extensions/amp-position-observer/0.1/amp-position-observer.js

```diff
@@ -193,6 +209,10 @@ export class AmpVisibilityObserver extends AMP.BaseElement {
       return;
     }
 
+    if (this.useAnimationWorklet_) {
```

Comment:
this won't work because `scroll` might be used with something that is not `amp-animation`. (e.g. `amp-glft`). My understanding is that the `scroll` triggered by `enter` and `exit` are messing things up. My suggestion adding another `bit` to the event object specifying whether it is coming from `enter or exit` and having worklet code in amp-animation ignore those.

this also made me realize the `maybeUninstallPositionObserver_` is also wrong and harder to implement than I thought. I suggest removing that code and putting a `TODO` instead. The todo can say
We want to remove position observer if scroll is only used by animationWorklet path of amp-animation. This involves having amp-animation signal back to us that it is using animationWorklet AND we need to ensure nothing else other than amp-animation is using scroll AND that `enter` and `exit`  are not used.

Reply:
If I understood you correctly, we want to only turn to the AnimationWorklet codepath if the `triggerScroll_()` is being called while the element is visible, not when it enters, exits the screen. That makes sense. I have implemented that here. TAL?

Also added the TODO will fix this in a separate PR.